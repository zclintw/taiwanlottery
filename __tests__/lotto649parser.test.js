const { promises: fs } = require('fs');

const lotto649parser = require('../lotto649parser');

describe('lotto649', () => {
  test('parse latest page', async () => {
    const html = await fs.readFile(`${__dirname}/mockdata/latest`, 'utf-8');

    expect.assertions(1);
    expect(lotto649parser(html)).toMatchSnapshot();
  });

  test('parse history page', async () => {
    const html = await fs.readFile(`${__dirname}/mockdata/history`, 'utf-8');

    expect.assertions(1);
    expect(lotto649parser(html)).toMatchSnapshot();
  });

  test('parse period page', async () => {
    const html = await fs.readFile(`${__dirname}/mockdata/givenperiod`, 'utf-8');

    expect.assertions(1);
    expect(lotto649parser(html)).toMatchSnapshot();
  });
});
