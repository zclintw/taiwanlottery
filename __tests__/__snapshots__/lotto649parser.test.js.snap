// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lotto649 parse history page 1`] = `
Array [
  Object {
    "date": "107/09/28",
    "nums": Array [
      1,
      2,
      7,
      20,
      27,
      37,
    ],
    "spl": 19,
    "term": "107000082",
  },
  Object {
    "date": "107/09/25",
    "nums": Array [
      19,
      25,
      27,
      39,
      40,
      46,
    ],
    "spl": 6,
    "term": "107000081",
  },
  Object {
    "date": "107/09/21",
    "nums": Array [
      3,
      5,
      10,
      11,
      29,
      45,
    ],
    "spl": 25,
    "term": "107000080",
  },
  Object {
    "date": "107/09/18",
    "nums": Array [
      11,
      23,
      34,
      36,
      41,
      45,
    ],
    "spl": 43,
    "term": "107000079",
  },
  Object {
    "date": "107/09/14",
    "nums": Array [
      19,
      29,
      36,
      37,
      45,
      46,
    ],
    "spl": 18,
    "term": "107000078",
  },
  Object {
    "date": "107/09/11",
    "nums": Array [
      14,
      16,
      18,
      23,
      25,
      44,
    ],
    "spl": 41,
    "term": "107000077",
  },
  Object {
    "date": "107/09/07",
    "nums": Array [
      2,
      8,
      14,
      21,
      37,
      44,
    ],
    "spl": 49,
    "term": "107000076",
  },
  Object {
    "date": "107/09/04",
    "nums": Array [
      4,
      6,
      10,
      15,
      42,
      49,
    ],
    "spl": 12,
    "term": "107000075",
  },
]
`;

exports[`lotto649 parse latest page 1`] = `
Array [
  Object {
    "date": "109/07/03",
    "nums": Array [
      13,
      19,
      20,
      24,
      40,
      45,
    ],
    "spl": 33,
    "term": "109000061",
  },
  Object {
    "date": "109/06/30",
    "nums": Array [
      26,
      29,
      31,
      43,
      46,
      49,
    ],
    "spl": 6,
    "term": "109000060",
  },
  Object {
    "date": "109/06/26",
    "nums": Array [
      10,
      12,
      16,
      37,
      40,
      41,
    ],
    "spl": 5,
    "term": "109000059",
  },
  Object {
    "date": "109/06/23",
    "nums": Array [
      1,
      2,
      16,
      22,
      27,
      45,
    ],
    "spl": 13,
    "term": "109000058",
  },
  Object {
    "date": "109/06/19",
    "nums": Array [
      1,
      15,
      16,
      25,
      43,
      46,
    ],
    "spl": 6,
    "term": "109000057",
  },
  Object {
    "date": "109/06/16",
    "nums": Array [
      6,
      13,
      20,
      23,
      29,
      46,
    ],
    "spl": 42,
    "term": "109000056",
  },
  Object {
    "date": "109/06/12",
    "nums": Array [
      1,
      6,
      28,
      30,
      33,
      48,
    ],
    "spl": 27,
    "term": "109000055",
  },
  Object {
    "date": "109/06/09",
    "nums": Array [
      8,
      17,
      28,
      35,
      36,
      47,
    ],
    "spl": 41,
    "term": "109000054",
  },
  Object {
    "date": "109/06/05",
    "nums": Array [
      3,
      5,
      8,
      24,
      26,
      49,
    ],
    "spl": 19,
    "term": "109000053",
  },
  Object {
    "date": "109/06/02",
    "nums": Array [
      1,
      6,
      22,
      25,
      33,
      48,
    ],
    "spl": 19,
    "term": "109000052",
  },
]
`;

exports[`lotto649 parse period page 1`] = `
Array [
  Object {
    "date": "109/06/19",
    "nums": Array [
      1,
      15,
      16,
      25,
      43,
      46,
    ],
    "spl": 6,
    "term": "109000057",
  },
]
`;
